weasyl/login.py

```diff
@@ -174,27 +174,33 @@ def create(form):
                    "weasyladmin", "weasylmod", "staff", "security"]:
         raise WeasylError("usernameInvalid")
     if email_exists(email):
-        raise WeasylError("emailExists")
+        # Prohibit attempting to determine if an email address is registered
+        email_unused = False
+    else:
+        # Email is unused, so the account can be created
+        email_unused = True
     if username_exists(sysname):
         raise WeasylError("usernameExists")
 
-    # Create pending account
-    token = security.generate_key(40)
-
-    d.engine.execute(d.meta.tables["logincreate"].insert(), {
-        "token": token,
-        "username": username,
-        "login_name": sysname,
-        "hashpass": passhash(password),
-        "email": email,
-        "birthday": birthday,
-        "unixtime": arrow.now(),
-    })
-
-    # Queue verification email
-    emailer.append([email], None, "Weasyl Account Creation", d.render(
-        "email/verify_account.html", [token, sysname]))
-    d.metric('increment', 'createdusers')
+    # Only attempt to create the account if the email is unused
+    if email_unused:
```

Comment:
For user-friendliness (and to resist timing attacks…? okay, that’s not very thorough), I think this should always send an e-mail, just with a different body to the effect of: “This e-mail address was used to sign up with the username …, but you already have a Weasyl account named …. You can \[reset your password] or \[request a username change].”

It would also be nice if this could always send exactly one insert query and rely on constraints to check for duplicate usernames and e-mail addresses, but that’s not how `logincreate` works yet.

(On including the existing username in this message: even though we require both username and e-mail address for a password reset and provide no “forgotten username” feature, I’m sure staff would respond to a request for a username sent from the address that registered it anyway.)

Reply:
>  [request a username change]
We don't currently expose the 'premium' function for username aliases to the public, though; would this just be emailing ``support@wea...``? Such as, "If you were attempting to change your username, please send an email from your registered email address for your Weasyl account, ${username}, to support@wea... and request that your username be changed."

Reply:
`[request a username change](mailto:support@weasyl.com)` was the idea, but we’ll probably want to have a whole discussion about wording.